spec/unit/shoryuken/error_handler_spec.rb

```diff
@@ -0,0 +1,53 @@
+require 'spec_helper'
+require 'airbrake/shoryuken/error_handler'
+
+module Airbrake
+  ##
+  # Provides tests for the Shoryuken integration.
+  module Shoryuken
+    RSpec.describe ErrorHandler do
+      let(:error) { AirbrakeTestError.new('shoryuken error') }
+      let(:body) { { message: 'message' } }
+      let(:notice) { double 'Notice' }
+
+      context "when there's an error" do
+        it 'notifies' do
+          expect(Airbrake).to receive(:build_notice).with(error, body: body).
+            and_return(notice)
+
+          expect(Airbrake).to receive(:notify).with(notice)
+
+          expect do
+            subject.call(nil, nil, nil, body) { raise error }
+          end.to raise_error(error)
+        end
+
+        context "and it's a batch" do
+          let(:body) { [{ message1: 'message1' }, { message2: 'message2' }] }
+
+          it 'notifies' do
+            expect(Airbrake).to receive(:build_notice).with(error, batch: body).
+              and_return(notice)
+
+            expect(Airbrake).to receive(:notify).with(notice)
+
+            expect do
+              subject.call(nil, nil, nil, body) { raise error }
+            end.to raise_error(error)
+          end
+        end
+      end
+
+      context "when there's no error" do
```

Comment:
I think this scenario is unnecessary to test. Instead, it would be nice to test the scenario when Airbrake is not configured: https://github.com/airbrake/airbrake/blob/c45bd6e0b4955dd0a0ad7fa87590383cc65fd8a4/spec/unit/sidekiq/error_handler_spec.rb#L37